src/python/WMCore/Services/TagCollector/TagCollector.py

```diff
@@ -112,3 +112,38 @@ def releases_by_architecture(self):
                 releases.add(item['label'])
             arch_dict[row['name']].extend(list(releases))
         return dict(arch_dict)
+
+
+    def defaultMicroArchVersionNumberByRelease(self, default_microarch=2):
+        """
+        Yield default microarchitecture by CMS release known in tag collector.
+        :return: dictionary ("release": default_microarch (or default to "2" if not found)).
+        """
+        rel_microarchs = {}
+        for row in self.data():
+            for item in row['project']:
+                microarch = item.get('default_micro_arch', default_microarch)
+                if isinstance(microarch, str):
+                    # format of this value is "arch-vN"
+                    microarch = int((microarch.split("-")[-1]).lstrip("v"))
+                rel_microarchs[item['label']] = microarch
+        return rel_microarchs
+
+    def getGreaterMicroarchVersionNumber(self, releases=None, default_microarch=2):
```

Comment:
Kenyi, please create a unit test for this new method and also update the docstring according to the contribution guidelines: https://github.com/dmwm/WMCore/blob/master/CONTRIBUTING.rst#project-docstrings-best-practices